src/distributions/mod.rs

```diff
@@ -82,6 +83,8 @@ pub struct Weighted<T> {
     pub item: T,
 }
 
+impl <T: Copy> Copy for Weighted<T> {}
```

Comment:
I didn't realise that it added the `Copy` bound in the impl! Thanks for that, I thought it added the bound to the struct definition (which would be weird).